### PR TITLE
Added randomized selection of set spawn points

### PR DESCRIPTION
**Pull request by kdreese**

### CODE REVIEW

File: src/states/Game.gd [hunk 1]
```diff
@@ -138,6 +145,32 @@ remote func spawn_peer_player(player_id: int) -> void:
...
+	var spawn_transforms_available := []
+	var num_open := 0
+	for p in spawn_transforms:
+		var num_adj_players := 0
+		for player in get_tree().get_nodes_in_group("Players"):
+			if player == my_player:
+				continue
+			if player.translation.distance_to(p.translation) < SPAWN_CAMP_REPELLANT_RADIUS:
+				num_adj_players += 1
+		if num_adj_players <= 0:
```
Comment (admoore0) [inline src/states/Game.gd]: This can just be ==

File: src/states/Game.gd [hunk 2]
```diff
@@ -138,6 +145,32 @@ remote func spawn_peer_player(player_id: int) -> void:
 		sync_targets(player_id)
 
 
+func set_spawn_point(my_player) -> void:
+	var i := 0
```
Comment (admoore0) [inline src/states/Game.gd]: Is this variable actually being used somewhere?

File: src/states/Game.gd [hunk 3]
```diff
@@ -138,6 +145,32 @@ remote func spawn_peer_player(player_id: int) -> void:
...
+	var spawn_transforms_available := []
+	var num_open := 0
```
Comment (admoore0) [inline src/states/Game.gd]: Add comments to all these declarations.

File: src/states/Game.gd [hunk 4]
```diff
@@ -138,6 +145,32 @@ remote func spawn_peer_player(player_id: int) -> void:
...
+		i += 1
+	var rand_spawn: Position3D
+	if num_open > 0:
+		var rand := randi() % num_open
+		rand_spawn = spawn_transforms_available[rand]
+	else:
+		push_warning("Couldn't find available spawn point")
+		rand_spawn = spawn_transforms[randi() % len(spawn_transforms)]
+	my_player.translation = rand_spawn.translation
+	my_player.rotation = rand_spawn.rotation
```
Comment (admoore0) [inline src/states/Game.gd]: Why not just use transform?

File: src/states/Game.gd [hunk 5]
```diff
@@ -1,12 +1,16 @@
 extends Node
 
 
+const SPAWN_CAMP_REPELLANT_RADIUS := 3
```
Comment (admoore0) [inline src/states/Game.gd]: maybe SPAWN_DISABLE_RADIUS?

File: src/states/Game.gd [hunk 6]
```diff
@@ -17,6 +21,7 @@ func _ready() -> void:
 
 	var curr_level := preload("res://src/levels/Level.tscn").instance() as Spatial
 	add_child(curr_level)
+	spawn_transforms = get_tree().get_nodes_in_group("SpawnPoints")
```
Comment (admoore0) [inline src/states/Game.gd]: This needs to have a .transform added, no?

**Review by admoore0 — CHANGES_REQUESTED** [review]

**Review by thearst3rd — APPROVED** [review]

**Review by admoore0 — APPROVED** [review]

**Pull request merged**